File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/inss/sobresalarios/OcorrenciaDeInssAtualizacao.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  javax.persistence.Column
 *  javax.persistence.MappedSuperclass
 *  javax.persistence.Temporal
 *  javax.persistence.TemporalType
 *  javax.persistence.Transient
 *  org.hibernate.annotations.Type
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios;

import br.jus.trt8.pjecalc.base.comum.Competencia;
import br.jus.trt8.pjecalc.base.comum.EntidadeBase;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.RepositorioDeOcorrenciaDeInssAtualizacao;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.TotalizadorOcorrenciaDeInssAtualizacao;
import java.math.BigDecimal;
import java.util.Collection;
import java.util.Date;
import javax.persistence.Column;
import javax.persistence.MappedSuperclass;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.persistence.Transient;
import org.hibernate.annotations.Type;

@MappedSuperclass
public abstract class OcorrenciaDeInssAtualizacao
extends EntidadeBase
implements Comparable<OcorrenciaDeInssAtualizacao> {
    private static final long serialVersionUID = 963715969685692563L;
    @Temporal(value=TemporalType.DATE)
    @Column(name="DDTINICIAL", nullable=false)
    private Date dataInicioPeriodo;
    @Temporal(value=TemporalType.DATE)
    @Column(name="DDTFINAL", nullable=false)
    private Date dataTerminoPeriodo;
    @Temporal(value=TemporalType.DATE)
    @Column(name="DDTOCORRENCIAINSS", nullable=false)
    private Date dataOcorrenciaInss;
    @Temporal(value=TemporalType.DATE)
    @Column(name="DDTEVENTO", nullable=false)
    private Date dataEvento;
    @Column(name="RVLDEVIDO", precision=19, scale=2)
    private BigDecimal devido = BigDecimal.ZERO;
    @Column(name="RVLINDICECORRECAO", precision=38, scale=25)
    private BigDecimal indiceCorrecao = BigDecimal.ZERO;
    @Column(name="RVLDEVIDOCORRIGIDO", precision=19, scale=2)
    private BigDecimal devidoCorrigido = BigDecimal.ZERO;
    @Column(name="RVLJUROS", precision=19, scale=2)
    private BigDecimal juros = BigDecimal.ZERO;
    @Column(name="RVLMULTA", precision=19, scale=2)
    private BigDecimal multa = BigDecimal.ZERO;
    @Column(name="RVLTOTAL", precision=19, scale=2)
    private BigDecimal total = BigDecimal.ZERO;
    @Column(name="RVLPAGO", precision=19, scale=2)
    private BigDecimal pago = BigDecimal.ZERO;
    @Column(name="RVLDEVIDODIFERENCA", precision=19, scale=2)
    private BigDecimal devidoDiferenca = BigDecimal.ZERO;
    @Column(name="RVLJUROSDIFERENCA", precision=19, scale=2)
    private BigDecimal jurosDiferenca = BigDecimal.ZERO;
    @Column(name="RVLMULTADIFERENCA", precision=19, scale=2)
    private BigDecimal multaDiferenca = BigDecimal.ZERO;
    @Column(name="RVLTOTALDIFERENCA", precision=19, scale=2)
    private BigDecimal totalDiferenca = BigDecimal.ZERO;
    @Column(name="SFLOCORRENCIADECIMOTERCEIRO", nullable=false, columnDefinition="VARCHAR2(1)")
    @Type(type="br.jus.trt8.pjecalc.base.comum.BooleanUserType")
    private Boolean ocorrenciaDecimoTerceiro = Boolean.FALSE;
    @Transient
    private boolean amortizado = false;
    @Transient
    private boolean parcialmenteAmortizado = false;

    public OcorrenciaDeInssAtualizacao() {
    }

    @Transient
    public String getCompetenciaFormatada() {
        Competencia competencia = Competencia.getInstance(this.getDataOcorrenciaInss());
        return competencia.getMes() + 1 + "/" + competencia.getAno();
    }

    public OcorrenciaDeInssAtualizacao(Class<? extends RepositorioDeOcorrenciaDeInssAtualizacao<? extends OcorrenciaDeInssAtualizacao>> classeDoRepositorio) {
        super(classeDoRepositorio);
    }

    public static TotalizadorOcorrenciaDeInssAtualizacao getTotalizador(Collection<? extends OcorrenciaDeInssAtualizacao> ocorrencias) {
        return new TotalizadorOcorrenciaDeInssAtualizacao(ocorrencias);
    }

    @Override
    public int compareTo(OcorrenciaDeInssAtualizacao o) {
        return this.getDataOcorrenciaInss().compareTo(o.getDataOcorrenciaInss());
    }

    public Date getDataInicioPeriodo() {
        return this.dataInicioPeriodo;
    }

    public void setDataInicioPeriodo(Date dataInicioPeriodo) {
        this.dataInicioPeriodo = dataInicioPeriodo;
    }

    public Date getDataTerminoPeriodo() {
        return this.dataTerminoPeriodo;
    }

    public void setDataTerminoPeriodo(Date dataTerminoPeriodo) {
        this.dataTerminoPeriodo = dataTerminoPeriodo;
    }

    public Date getDataOcorrenciaInss() {
        return this.dataOcorrenciaInss;
    }

    public void setDataOcorrenciaInss(Date dataOcorrenciaInss) {
        this.dataOcorrenciaInss = dataOcorrenciaInss;
    }

    public Date getDataEvento() {
        return this.dataEvento;
    }

    public void setDataEvento(Date dataEvento) {
        this.dataEvento = dataEvento;
    }

    public BigDecimal getDevido() {
        return this.devido;
    }

    public void setDevido(BigDecimal devido) {
        this.devido = devido;
    }

    public BigDecimal getIndiceCorrecao() {
        return this.indiceCorrecao;
    }

    public void setIndiceCorrecao(BigDecimal indiceCorrecao) {
        this.indiceCorrecao = indiceCorrecao;
    }

    public BigDecimal getDevidoCorrigido() {
        return this.devidoCorrigido;
    }

    public void setDevidoCorrigido(BigDecimal devidoCorrigido) {
        this.devidoCorrigido = devidoCorrigido;
    }

    public BigDecimal getJuros() {
        return this.juros;
    }

    public void setJuros(BigDecimal juros) {
        this.juros = juros;
    }

    public BigDecimal getMulta() {
        return this.multa;
    }

    public void setMulta(BigDecimal multa) {
        this.multa = multa;
    }

    public BigDecimal getTotal() {
        return this.total;
    }

    public void setTotal(BigDecimal total) {
        this.total = total;
    }

    public BigDecimal getPago() {
        return this.pago;
    }

    public void setPago(BigDecimal pago) {
        this.pago = pago;
    }

    public BigDecimal getDevidoDiferenca() {
        return this.devidoDiferenca;
    }

    public void setDevidoDiferenca(BigDecimal devidoDiferenca) {
        this.devidoDiferenca = devidoDiferenca;
    }

    public BigDecimal getJurosDiferenca() {
        return this.jurosDiferenca;
    }

    public void setJurosDiferenca(BigDecimal jurosDiferenca) {
        this.jurosDiferenca = jurosDiferenca;
    }

    public BigDecimal getMultaDiferenca() {
        return this.multaDiferenca;
    }

    public void setMultaDiferenca(BigDecimal multaDiferenca) {
        this.multaDiferenca = multaDiferenca;
    }

    public BigDecimal getTotalDiferenca() {
        return this.totalDiferenca;
    }

    public void setTotalDiferenca(BigDecimal totalDiferenca) {
        this.totalDiferenca = totalDiferenca;
    }

    public boolean isAmortizado() {
        return this.amortizado;
    }

    public void setAmortizado(boolean amortizado) {
        this.amortizado = amortizado;
    }

    public boolean isParcialmenteAmortizado() {
        return this.parcialmenteAmortizado;
    }

    public void setParcialmenteAmortizado(boolean parcialmenteAmortizado) {
        this.parcialmenteAmortizado = parcialmenteAmortizado;
    }

    public Boolean getOcorrenciaDecimoTerceiro() {
        return this.ocorrenciaDecimoTerceiro;
    }

    public void setOcorrenciaDecimoTerceiro(Boolean ocorrenciaDecimoTerceiro) {
        this.ocorrenciaDecimoTerceiro = ocorrenciaDecimoTerceiro;
    }
}

File: src/lib/pjecalc/core/dominio/calculo/inss/sobresalarios/ocorrencia-de-inss-atualizacao.ts
/**
 * PJe-Calc v2.15.1 — OcorrenciaDeInssAtualizacao (abstract)
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.OcorrenciaDeInssAtualizacao
 *
 * Ref Java: pjecalc-fonte/.../sobresalarios/OcorrenciaDeInssAtualizacao.java
 *
 * Linha-mensal de atualização de INSS pós-liquidação. Diferente de
 * `OcorrenciaDeInss`: não armazena alíquotas ou base, apenas totais já
 * calculados (devido, juros, multa, total, pago, diferenças).
 *
 * Subclasses concretas:
 *   OcorrenciaDeInssSobreSalariosDevidosAtualizacao
 *   OcorrenciaDeInssSobreSalariosPagosAtualizacao
 */
import Decimal from 'decimal.js';

const ZERO = new Decimal(0);

export abstract class OcorrenciaDeInssAtualizacao {
  protected dataInicioPeriodo: Date | null = null;
  protected dataTerminoPeriodo: Date | null = null;
  protected dataOcorrenciaInss: Date | null = null;
  protected dataEvento: Date | null = null;
  protected devido: Decimal = ZERO;
  protected indiceCorrecao: Decimal = ZERO;
  protected devidoCorrigido: Decimal = ZERO;
  protected juros: Decimal = ZERO;
  protected multa: Decimal = ZERO;
  protected total: Decimal = ZERO;
  protected pago: Decimal = ZERO;
  protected devidoDiferenca: Decimal = ZERO;
  protected jurosDiferenca: Decimal = ZERO;
  protected multaDiferenca: Decimal = ZERO;
  protected totalDiferenca: Decimal = ZERO;
  protected ocorrenciaDecimoTerceiro: boolean = false;
  // transient
  protected amortizado: boolean = false;
  protected parcialmenteAmortizado: boolean = false;

  getDataInicioPeriodo(): Date | null { return this.dataInicioPeriodo; }
  setDataInicioPeriodo(d: Date | null): void { this.dataInicioPeriodo = d; }

  getDataTerminoPeriodo(): Date | null { return this.dataTerminoPeriodo; }
  setDataTerminoPeriodo(d: Date | null): void { this.dataTerminoPeriodo = d; }

  getDataOcorrenciaInss(): Date | null { return this.dataOcorrenciaInss; }
  setDataOcorrenciaInss(d: Date | null): void { this.dataOcorrenciaInss = d; }

  getDataEvento(): Date | null { return this.dataEvento; }
  setDataEvento(d: Date | null): void { this.dataEvento = d; }

  getDevido(): Decimal { return this.devido; }
  setDevido(v: Decimal): void { this.devido = v; }

  getIndiceCorrecao(): Decimal { return this.indiceCorrecao; }
  setIndiceCorrecao(v: Decimal): void { this.indiceCorrecao = v; }

  getDevidoCorrigido(): Decimal { return this.devidoCorrigido; }
  setDevidoCorrigido(v: Decimal): void { this.devidoCorrigido = v; }

  getJuros(): Decimal { return this.juros; }
  setJuros(v: Decimal): void { this.juros = v; }

  getMulta(): Decimal { return this.multa; }
  setMulta(v: Decimal): void { this.multa = v; }

  getTotal(): Decimal { return this.total; }
  setTotal(v: Decimal): void { this.total = v; }

  getPago(): Decimal { return this.pago; }
  setPago(v: Decimal): void { this.pago = v; }

  getDevidoDiferenca(): Decimal { return this.devidoDiferenca; }
  setDevidoDiferenca(v: Decimal): void { this.devidoDiferenca = v; }

  getJurosDiferenca(): Decimal { return this.jurosDiferenca; }
  setJurosDiferenca(v: Decimal): void { this.jurosDiferenca = v; }

  getMultaDiferenca(): Decimal { return this.multaDiferenca; }
  setMultaDiferenca(v: Decimal): void { this.multaDiferenca = v; }

  getTotalDiferenca(): Decimal { return this.totalDiferenca; }
  setTotalDiferenca(v: Decimal): void { this.totalDiferenca = v; }

  isAmortizado(): boolean { return this.amortizado; }
  setAmortizado(v: boolean): void { this.amortizado = v; }

  isParcialmenteAmortizado(): boolean { return this.parcialmenteAmortizado; }
  setParcialmenteAmortizado(v: boolean): void { this.parcialmenteAmortizado = v; }

  getOcorrenciaDecimoTerceiro(): boolean { return this.ocorrenciaDecimoTerceiro; }
  setOcorrenciaDecimoTerceiro(v: boolean): void { this.ocorrenciaDecimoTerceiro = v; }

  /**
   * getCompetenciaFormatada (Java linha 79) — retorna "MM/YYYY".
   * Obs: Java soma 1 ao mês (Competencia.getMes() é 0-based lá); aqui o mês já
   * é 1-based, então retornamos direto.
   */
  getCompetenciaFormatada(): string {
    if (!this.dataOcorrenciaInss) return '';
    const mes = this.dataOcorrenciaInss.getMonth() + 1;
    const ano = this.dataOcorrenciaInss.getFullYear();
    return `${mes}/${ano}`;
  }

  /** compareTo — por dataOcorrenciaInss (Java linha 93) */
  compareTo(o: OcorrenciaDeInssAtualizacao): number {
    const a = this.dataOcorrenciaInss;
    const b = o.dataOcorrenciaInss;
    if (!a || !b) return 0;
    return a.getTime() - b.getTime();
  }
}
